examples/node/Examples/read-and-write-data.js

```diff
@@ -0,0 +1,611 @@
+        name: "Moe Chughtai",
+      });
+      person2 = realm.create("Person", {
+        name: "Chris Bush",
+      });
+      dog1 = realm.create("Dog", {
+        name: "Barky",
+        owner: person1,
+      });
+      dog2 = realm.create("Dog", {
+        name: "Fido",
+        owner: person2,
+      });
+    });
+
+    // :code-block-start: read-and-write-sorted-queries
+    // retrieve the set of Task objects
+    const tasks = realm.objects("Task");
+    // Sort tasks by name in ascending order
+    const tasksByName = tasks.sorted("name");
+    // Sort tasks by name in descending order
+    const tasksByNameDescending = tasks.sorted("name", true);
+    // Sort tasks by priority in descending order and then by name alphabetically
+    const tasksByPriorityDescendingAndName = tasks.sorted([
+      ["priority", true],
+      ["name", false],
+    ]);
+    // Sort dogs by dog's owner's name.
+    let dogsByOwnersName = realm.objects("Dog").sorted("owner.name");
+    // :code-block-end:
+
+    // tasks sorted by name ascending
+    expect(tasksByName.map((myTask) => myTask.name)).toStrictEqual([
+      "fold the laundry",
+      "go grocery shopping",
+      "throw out the trash",
+      "wash and dry the laundry",
+    ]);
+
+    // tasks sorted by name descending
+    expect(tasksByNameDescending.map((myTask) => myTask.name)).toStrictEqual([
+      "wash and dry the laundry",
+      "throw out the trash",
+      "go grocery shopping",
+      "fold the laundry",
+    ]);
+
+    // tasks sorted by priority descending, and then by name ascending
+    expect(
+      tasksByPriorityDescendingAndName.map((myTask) => myTask.name)
+    ).toStrictEqual([
+      "go grocery shopping",
+      "throw out the trash",
+      "fold the laundry",
+      "wash and dry the laundry",
+    ]);
+    // dogs by dog's owner's name.
+    expect(JSON.stringify(dogsByOwnersName)).toBe(
+      JSON.stringify([
+        { name: "Fido", owner: { name: "Chris Bush", age: null }, age: null },
+        {
+          name: "Barky",
+          owner: { name: "Moe Chughtai", age: null },
+          age: null,
+        },
+      ])
+    );
+
+    // delete tasks, persons, and dogs
+    realm.write(() => {
+      realm.delete(task);
+      realm.delete(task2);
+      realm.delete(task3);
+      realm.delete(task4);
+
+      realm.delete(person1);
+      realm.delete(person2);
+
+      realm.delete(dog1);
+      realm.delete(dog2);
+    });
+
+    // Discard the references.
+    task = null;
+    task2 = null;
+    task3 = null;
+    task4 = null;
+
+    person1 = null;
+    person2 = null;
+
+    dog1 = null;
+    dog2 = null;
+    // close the realm
+    realm.close();
+  });
+  test("should write a new object", async () => {
+    // a realm is opened
+    const realm = await Realm.open({
+      path: "myrealm",
+      schema: [DogSchema, PersonSchema],
+    });
+    // :code-block-start: read-and-write-create-a-new-object
+    // Declare the instance.
```

Comment:
We could be more precise with our wording here:
```suggestion
    // Declare the variable that will hold the dog instance.
```